PWGLF/SPECTRA/ChargedHadrons/dNdPt/Framework/AliAnalysisTaskMKBase.cxx

```diff
@@ -1021,6 +1023,10 @@ Bool_t AliAnalysisTaskMKBase::InitMCParticle()
     fMCPrimSec = 2;
     fMCProdcutionType = AlidNdPtTools::kSecMaterial;
   }
+  if(fMCPileUpTrack){
+    fMCPrimSec = 3;
+    fMCProdcutionType = AlidNdPtTools::ProductionType::kPileUpTrack;
+  }
```

Comment:
maybe it is better to make lines 1014 - 1029 an if, else if, else if statement and add an else clause setting the two variables to -1 or something similar.
Otherwise fMCPrimSec and fMCProdcutionType could be uninitialized.

Reply:
okay thinking about it again 'else if' might be a bad idea since fMCisPrim is probably also true for pileup (as we have seen in your efficiency plots) but maybe we could initialize the variables with something like this before the if clause:
fMCPrimSec = -1;
fMCProdcutionType = AlidNdPtTools::ProductionType::kUnknown;